src/vl-map-features-layer.js

```diff
@@ -0,0 +1,186 @@
+   */
+  set features(features) {
+    this.setAttribute('features', JSON.stringify(features));
+  }
+
+  get _autoExtent() {
+    return this.getAttribute('auto-extent') != undefined;
+  }
+
+  get _autoExtentMaxZoom() {
+    return this.getAttribute('auto-extent-max-zoom');
+  }
+
+  get cluster() {
+    return this.getAttribute('cluster') != undefined;
+  }
+
+  get _clusterDistance() {
+    return this.getAttribute('cluster-distance');
+  }
+
+  /**
+   * Verwijdert de stijl van al de kaartlaag features.
+   */
+  removeFeaturesStyle() {
+    if (this.source && this.source.getFeatures()) {
+      this.source.getFeatures().forEach((feature) => {
+        feature.setStyle(null);
+      });
+    }
+  }
+
+  /**
+   * Geeft de feature terug op basis van het id attribuut.
+   *
+   * @param {number} id
+   * @return {Object}
+   */
+  getFeature(id) {
+    if (this.source && this.source.getFeatures()) {
+      return this.source.getFeatures().filter((feature) => {
+        return feature.getId() === id;
+      })[0];
+    }
+  }
+
+  /**
+   * Geeft de cluster terug op basis van het id attribuut.
+   *
+   * @param {number} id
+   * @return {boolean}
+   */
+  getCluster(id) {
+    if (this._layer) {
+      return this._layer.getSource().getFeatures().filter((cluster) => {
+        const features = cluster.get('features');
+        if (features) {
+          return features.some((feature) => {
+            return feature.getId() === id;
+          });
+        } else {
+          return false;
+        }
+      })[0];
+    }
+  }
+
+  /**
+   * Zoom naar alle features in deze layer.
+   *
+   * @param {number} maxZoom - Hoe diep er maximaal ingezoomd mag worden.
+   */
+  async zoomToExtent(maxZoom) {
+    this.mapElement.zoomTo(this.__boundingBox, maxZoom);
+  }
+
+  _autoExtentChangedCallback() {
+    this._autoZoomToExtent();
+  }
+
+  _featuresChangedCallback(oldValue, newValue) {
+    if (newValue && this._layer) {
+      this.source.clear();
+      this.source.addFeatures(this.features);
+      this._autoZoomToExtent();
+      this.rerender();
+    }
+  }
+
+  _autoZoomToExtent() {
+    if (this._autoExtent) {
+      this.zoomToExtent(this._autoExtentMaxZoom);
+    }
+  }
+
+  get __boundingBox() {
+    if (this.source && this.source.getFeatures().length > 0) {
+      return this.source.getExtent();
+    }
+  }
+
+  __createSource() {
+    const source = new OlVectorSource({
+      features: this.features,
+    });
+    return this.cluster ? this.__createClusterSource(source) : source;
+  }
+
+  __createClusterSource(source) {
+    return new OlClusterSource({
+      distance: this._clusterDistance,
+      source: source,
+      geometryFunction: (feature) => {
+        const geometry = feature.getGeometry();
+        if (geometry instanceof OlPoint) {
+          return geometry;
+        } else {
+          return this.__ignoreClustering();
+        }
+      },
+    });
+  }
+
+  __ignoreClustering() {
+    return null;
```

Comment:
deze is bedoeld voor toekomstige implementatie vermoed ik

Reply:
Tis nog uit 'oude' code, maar ik denk dat dat vooral bedoeld is om het te doen lezen als een boek. Dus zodat je in __createClusterSource kunt zien staan: ignoreClustering ipv null.